### PR TITLE
[feat][doc] Add Pulsar Release Note Guide

### PR DESCRIPTION
**Pull request by Anonymitaet**
This PR adds workflow for "How to submit Pulsar release notes".

(Move content from the [google doc](https://docs.google.com/document/d/1cwNkBefKyV6OPbEXnUrcCdVZi0i2BezqL6vAL7VqVC0/edit?pli=1#heading=h.lafhic4xpni8) to wiki.)

CC @urfreespace @SignorMercurio 

The workflow of C++ and Python might be changed according to https://github.com/apache/pulsar/issues/17918#issuecomment-1273075920

- [x] `doc`

### CODE REVIEW

**Comment by Anonymitaet** [comment]
@BewareMyPower @momo-jun @tisonkun @D-2-Ed @DaveDuggins could you please review this PR? Thank you!

File: wiki/release/release-process.md [hunk 1]
```diff
@@ -454,7 +454,7 @@ Send out a PR request for review.
 
 ## Write release notes
 
-See [Pulsar Release Notes Guide](https://docs.google.com/document/d/1cwNkBefKyV6OPbEXnUrcCdVZi0i2BezqL6vAL7VqVC0/edit#).
+See [Pulsar Release Notes Guide](../release/release-note-guide.md).
```
Comment (tisonkun) [inline wiki/release/release-process.md:457]:
```suggestion
See [Pulsar Release Notes Guide](../release-note-guide.md).
```

File: wiki/release/release-note-guide.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    </table>
+
+    After the new release is published, all the information about the release is automatically added to the [Pulsar Release Note page](https://pulsar.apache.org/release-notes/).
+
+2. Check whether the release information is shown on the [Pulsar Release Note page](https://pulsar.apache.org/release-notes/) after the website is updated and built successfully.
+
+
+
+
+
```
Comment (tisonkun) [inline wiki/release/release-note-guide.md:69]:
```suggestion

```

File: wiki/release/release-note-guide.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        <td rowspan="2">Pulsar clients</td>
+        <td> - Java<br><br> - WebSocket<br><br> - C++<br><br> - Python</td>
+        <td>Add separate release notes for them, that is, create independent sections in the release note.<br><br>Example<br><br><img title="Java client release note example" alt="Java client release note example" src="../assets/release-note-guide-1.png"></td>
+      </tr>
+      <tr>
+        <td> - Go<br><br> - Node.js<br><br> - C#</td>
+        <td>No action is needed. You do not need to take care of them since their release notes are synced from their repos to the <a href="https://pulsar.apache.org/release-notes/">Pulsar Release Note page</a>.</td>
+      </tr>
+    </tbody>
+    </table>
```
Comment (tisonkun) [inline wiki/release/release-note-guide.md:60]: Perhaps transform to markdown tables?

File: wiki/release/release-note-guide.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
+This guide explains everything about Pulsar release notes.
```
Comment (tisonkun) [inline wiki/release/release-note-guide.md:1]:
```suggestion
# Release Note Guide

This guide explains everything about Pulsar release notes.
```

... and indent one level for all headings below.

**Review by tisonkun — APPROVED** [review]
Thanks for sending this patch! I think it's OK to add as is but still leave some comments XD.

File: wiki/release/release-note-guide.md (hunk 3 above)
Comment (Anonymitaet) [reply wiki/release/release-note-guide.md:60]: Markdown tables do not support complex format (eg. "merge cells"), so I use HTML instead.

File: wiki/release/release-note-guide.md (hunk 4 above)
Comment (Anonymitaet) [reply wiki/release/release-note-guide.md:3]: Good suggestion! I've incorporated and added a TOC

image

File: wiki/release/release-note-guide.md (hunk 3 above)
Comment (tisonkun) [reply wiki/release/release-note-guide.md:70]: Make sense.

File: wiki/release/release-process.md [hunk 5]
```diff
@@ -454,7 +454,7 @@ Send out a PR request for review.
 
 ## Write release notes
 
-See [Pulsar Release Notes Guide](https://docs.google.com/document/d/1cwNkBefKyV6OPbEXnUrcCdVZi0i2BezqL6vAL7VqVC0/edit#).
+See [Pulsar Release Notes Guide](../release-note-guide.md).
```
Comment (tisonkun) [inline wiki/release/release-process.md:457]:
```suggestion
See [Pulsar Release Notes Guide](./release-note-guide.md).
```

Sorry. The first comment makes a mistake >_<

File: wiki/release/release-note-guide.md (hunk 4 above)
Comment (tisonkun) [reply wiki/release/release-note-guide.md:3]: See my comment here https://github.com/apache/pulsar/pull/17470#discussion_r963277120 that on GitHub we don't need to manually maintain a TOC :)

File: wiki/release/release-process.md (hunk 1 above)
Comment (Anonymitaet) [reply wiki/release/release-process.md:457]: I use `../release-note-guide.md` but it goes to 404, so I use `../release/release-note-guide.md` which works well.

File: wiki/release/release-process.md (hunk 5 above)
Comment (Anonymitaet) [reply wiki/release/release-process.md]: No worries, I've updated it to `./release-note-guide.md`, which works well and is simple. Thank you!

**Review by tisonkun — COMMENTED** [review]
You can add `wiki/**` to https://github.com/apache/pulsar/blob/master/.github/changes-filter.yaml `docs` section so that we don't waste time to run full CI tasks.

**Comment by Anonymitaet** [comment]
> wiki/**

@tisonkun thanks for your reminder! Done in https://github.com/apache/pulsar/pull/17999/files#diff-1b2ef07bf2b24cfe900fb2d2b36e0ca604e96b38cd48e00c82d21d64a9c06697

**Pull request merged**